src/profile/helpers/editData.ts

```diff
@@ -0,0 +1,202 @@
+import _ from 'lodash';
+
+import {
+  ProfileRoot,
+  ProfileData,
+  AddressNode,
+  EmailNode,
+  PhoneNode,
+  PrimaryAddress,
+  PrimaryEmail,
+  PrimaryPhone,
+  Language,
+} from '../../graphql/typings';
+import { formConstants } from '../constants/formConstants';
+import getAddressesFromNode from '../helpers/getAddressesFromNode';
+import getEmailsFromNode from '../helpers/getEmailsFromNode';
+import getPhonesFromNode from '../helpers/getPhonesFromNode';
+
+export type BasicDataValue = Pick<
+  ProfileData,
+  'firstName' | 'nickname' | 'lastName'
+>;
+export type AdditionalInformationValue = {
+  language: Language;
+};
+export type AddressValue = Pick<
+  AddressNode,
+  'address' | 'city' | 'postalCode' | 'countryCode'
+>;
+
+export type EmailValue = Pick<EmailNode, 'email'>;
+export type PhoneValue = Pick<PhoneNode, 'phone'>;
+
+type BasicDataSource = BasicDataValue & { id: ProfileData['id'] };
+type AdditionalInformationSource = Pick<ProfileData, 'id' | 'language'>;
+export type EditDataProfileSource =
+  | BasicDataSource
+  | AdditionalInformationSource
+  | AddressNode
+  | PhoneNode
+  | EmailNode;
+
+export type MultiItemProfileNode = AddressNode | PhoneNode | EmailNode;
+
+export const basicDataType = 'basic-data';
+export const additionalInformationType = 'additional-information';
+
+export type EditDataType =
+  | 'phones'
+  | 'emails'
+  | 'addresses'
+  | typeof basicDataType
+  | typeof additionalInformationType;
```

Comment:
Typescript wont allow typos. Cant use 'adress' anywhere where EditDataType is expected.

basicDataType (and currently unused additionalInformationType) are defined as const because they are used also as strings.

Example in current code:
"const testId = basicDataType;"

(Yes, "const testId:EditDataType = 'basic-data';" is an alternative.) 

"Address" etc are always passed to func/params expecting EditDataType or just passed on as a "dataType" variable.